modules/store/migrations/13_0_1/index.ts

```diff
@@ -0,0 +1,89 @@
+import * as ts from 'typescript';
+import { Rule, chain, Tree } from '@angular-devkit/schematics';
+import {
+  visitTSSourceFiles,
+  RemoveChange,
+  commitChanges,
+  InsertChange,
+} from '../../schematics-core';
+
+function updateCreateSelectorGenerics(): Rule {
+  return (tree: Tree) => {
+    visitTSSourceFiles(tree, (sourceFile) => {
+      const runMigration = sourceFile.statements
+        .filter(ts.isImportDeclaration)
+        .filter(
+          (importDeclaration) =>
+            importDeclaration.moduleSpecifier.getText(sourceFile) ===
+              "'@ngrx/store'" ||
+            importDeclaration.moduleSpecifier.getText(sourceFile) ===
+              '"@ngrx/store"'
+        )
+        .some((importDeclaration) => {
+          return importDeclaration.importClause?.namedBindings
+            ?.getText(sourceFile)
+            .includes('createSelector');
+        });
+
+      if (!runMigration) return;
+
+      const changes: (RemoveChange | InsertChange)[] = [];
+      ts.forEachChild(sourceFile, crawl);
+      return commitChanges(tree, sourceFile.fileName, changes);
+
+      function crawl(node: ts.Node) {
+        ts.forEachChild(node, crawl);
+
+        if (!ts.isCallExpression(node)) return;
+
+        const { typeArguments } = node;
+
+        if (!typeArguments?.length) return;
+        const typeArgumentsLength = typeArguments?.length ?? 0;
+        if (typeArgumentsLength < 3) return;
+        if (!ts.isIdentifier(node.expression)) return;
+        if (node.expression.text !== 'createSelector') return;
+
+        const lastTypeArgumentIndex = typeArgumentsLength - 1;
+        const slicesTypeArguments = typeArguments.slice(
+          1,
+          lastTypeArgumentIndex
+        );
+        const updatedTypeArguments = [
+          typeArguments[0],
+          slicesTypeArguments,
+          typeArguments[lastTypeArgumentIndex],
+        ];
+
+        const isSelectorWithProps =
+          node.arguments.length === typeArguments.length - 2;
+        if (isSelectorWithProps) return;
```

Comment:
Should we log something for these cases?
I think the Angular migrations do this something, when a manual intervention is needed, which I liked.

Reply:
The selectors with props require no migration because their signatures have all been left unmodified.

Reply:
Oooooooh, right... 😅